src/cpp/session/resources/grid/gridviewer.js

```diff
@@ -611,6 +624,83 @@ var createFilterUI = function(idx, col) {
   return host;
 };
 
+var openColumnTypeChooserUI = function(idx, col, ele, onDismiss) {
+  invokeFilterPopup(ele, function(popup) {
+    var columnTypes = [
+      "logical", "integer", "double", "character", "date", "datetime", "numeric" 
```

Comment:
looks like it's missing some column types -- should we include `factor`, `time`, etc?